File: Back/NodeJS/modules/bets/src/infrastructure/Outbox/Outbox.ts
import { IDateTimeProvider } from "../../../../shared/domain/IDateTimeProvider";

export class Outbox {
    HandledAt?: Date;
    Handled(dateProvider: IDateTimeProvider) {
        this.HandledAt = dateProvider.GetDate()
    }
    constructor(public Id: string, public Type: string, public Data: string, dateProvider: IDateTimeProvider) {
        this.CreatedAt = dateProvider.GetDate()
    }

    CreatedAt: Date
}

File: Back/.Net/Modules/Shared/BetFriends.Shared.Infrastructure/Outboxes/Outbox.cs
using BetFriends.Shared.Application.Abstractions;

namespace BetFriends.Shared.Infrastructure.Outboxes;

public class Outbox
{
    public Outbox(Guid id, string type, string data, DateTime occurredOn)
    {
        Id = id;
        OccurredOn = occurredOn;
        Type = type;
        Data = data;
    }

    public Guid Id { get; }
    public DateTime OccurredOn { get; }
    public DateTime? ProcessedOn { get; private set; }
    public string Type { get; }
    public string Data { get; }

    public void Handled(IDateProvider dateProvider)
    {
        ProcessedOn = dateProvider.GetDate();
    }
}
